theano/tensor/nnet/abstract_conv.py

```diff
@@ -2097,10 +2168,20 @@ def perform(self, node, inp, out_):
         axes_order = (1, 0) + tuple(range(2, self.convdim + 2))
         flip_filters = ((slice(None), slice(None)) +
                         (slice(None, None, -1),) * self.convdim)
+
+        def correct_for_groups(mat):
```

Comment:
I haven't used this anywhere in the c code. The c code gives  more freedom than the python implementation.There was no need reshapes and  transposes .